qa-pipelines/tasks/lib/prepare-kubeconfig.sh

```diff
@@ -5,10 +5,44 @@ read_yaml_key() {
     ruby -r yaml -e "puts YAML.load_file('$1')[\"$2\"]"
 }
 
+aws_auth_cm_yaml() {
+    cat << 'EOF' >& 2
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: aws-auth
+      namespace: kube-system
+    data:
+      mapRoles: |
+        - rolearn: <ARN-node-instance-role>
+          username: system:node:{{EC2PrivateDNSName}}
+          groups:
+            - system:bootstrappers
+            - system:nodes
+        - rolearn: arn:aws:iam::138384977974:role/eksServiceRole
+          username: eksServiceRole
+          groups:
+            - system:masters
+EOF
+}
+
 pool_file=${pool_file:-pool.kube-hosts/metadata}
 mkdir -p /root/.kube
 if [[ $(read_yaml_key ${pool_file} kind) == "Config" ]]; then
     cp ${pool_file} /root/.kube/config
+    if cat ~/.kube/config | grep "eks.amazonaws.com" > /dev/null ; then
```

Comment:
useless use of cat

Reply:
Because EKS clusters will be created by users and then they will have to prepare their kubeconfig for eksServiceRole as in line 38

Reply:
OK, I'll take another look at this once there are comments explaining why that's needed.